roomserver/internal/input_membership.go

```diff
@@ -132,6 +131,15 @@ func updateMembership(
 	}
 }
 
+func (r *RoomserverInternalAPI) isLocalTarget(event *gomatrixserverlib.Event) bool {
+	targetLocal := false
```

Comment:
Naming: I find all this localTarget targetLocal stuff confusing because I don't know if it is saying "the target is local" or "target local users" - the former is informational and the latter is an instruction. `isTargetLocalUser(event)` would be a bit clearer.